Release docs for terraform version V1.4.0 (#243)
couchbasecloud/terraform-provider-couchbase-capella: docs/guides/1.4.0-upgrade-guide.md

---
layout: "couchbase-capella"
page_title: "Couchbase Capella Provider 1.4.0: Upgrade and Information Guide"
sidebar_current: "docs-couchbase-capella-guides-140-upgrade-guide"
description: |-
Couchbase Capella Provider 1.4.0: Upgrade and Information Guide
---

# Couchbase Capella Provider 1.4.0: Upgrade and Information Guide

> New features in 1.4.0
> 
> 
> ## New Features

Reviewer: Should this be after the heading?

```suggestion

## New Features

New features in 1.4.0:

```

Reviewer 2: +1 to @simon-dew  suggestion

Author: @simon-dew We can update it but in the past release we have done it like this only

New features in <Version Number> and in next line we have added "New Features" as heading text.

Attaching a screenshot of last release for reference. 
<img width="873" alt="Screenshot 2024-11-25 at 1 00 10 PM" src="https://github.com/user-attachments/assets/1fd63924-2aad-4032-b1bb-73f93ee798fb">

Reviewer: Fair enough! I'm happy to keep it as it is for consistency

* Enable Flush Bucket [`couchbase-capella_flush`](https://registry.terraform.io/providers/couchbasecloud/couchbase-capella/latest/docs/resources/flush)
* Enable Azure Network Peer [`couchbase-capella_network_peer`](https://registry.terraform.io/providers/couchbasecloud/couchbase-capella/latest/docs/resources/network_peer)

## Flush Bucket

> Use the new `flush` resource to flush a bucket

Reviewer: Full stop at end:

```suggestion
Use the new `flush` resource to flush a bucket.
```

    organization_id            = var.organization_id
    project_id                 = var.project_id
    cluster_id                 = var.cluster_id
    type                       = var.bucket.type
    storage_backend            = var.bucket.storage_backend
    memory_allocation_in_mb    = var.bucket.memory_allocation_in_mb
    bucket_conflict_resolution = var.bucket.bucket_conflict_resolution
    durability_level           = var.bucket.durability_level
    replicas                   = var.bucket.replicas
    flush                      = var.bucket.flush
    time_to_live_in_seconds    = var.bucket.time_to_live_in_seconds
    eviction_policy            = var.bucket.eviction_policy
}
```
```
output "new_flush" {
    value = couchbase-capella_flush.new_flush
}
```
```
resource "couchbase-capella_flush" "new_flush" {
    organization_id = var.organization_id
    project_id      = var.project_id
    cluster_id      = var.cluster_id
    bucket_id       = var.bucket_id
}
```
For more information, see the [flush bucket examples](https://github.com/couchbasecloud/terraform-provider-couchbase-capella/tree/main/examples/flush_bucket).

## Azure Network Peer

> Use the `azure_network_peer_command` data source to get an AWS command for configuring VPC endpoint.

Reviewer 2: ```suggestion
Use the `azure_network_peer_command` data source to get an Azure CLI command for configuring VNET endpoint.
```

General: I have not paid close attention but please double check to ensure no copy paste errors. @PaulomeeCb  @a-atri

> Use the `azure_network_peer_command` data source to get an Azure CLI command for configuring VNET endpoint.

Reviewer 3: Can we add a line about the Admin consent granting process like-
```suggestion
Use the `azure_network_peer_command` data source to get an Azure CLI command for configuring VNET endpoint.

Note:  Before using this API, please make sure that the Admin consent granting process has been completed through the Capella UI. For more information, please refer to the [steps here] (https://docs.couchbase.com/cloud/management-api-reference/index.html#tag/Network-Peers/operation/getAzureVnetPeeringCommand)
```

```
output "azure_network_peer_command" {
  value = data.couchbase-capella_azure_network_peer_command.azure_network_peer_command
}
```
```
data "couchbase-capella_azure_network_peer_command" "azure_network_peer_command" {
  organization_id                = var.organization_id
  project_id                     = var.project_id
  cluster_id                     = var.cluster_id
  tenant_id                      = var.tenant_id
  vnet_id                        = var.vnet_id
  subscription_id                = var.subscription_id
  resource_group                 = var.resource_group
  vnet_peering_service_principal = var.vnet_peering_service_principal
}
```

> Create a new Azure network peer, use the `network-peer` resource as follows:

Reviewer: ```suggestion
To create a new Azure network peer, use the `network-peer` resource as follows:
```
